src/config.ts

```diff
@@ -320,6 +330,20 @@ export class KubeConfig {
         });
     }
 
+    public exportConfig(): string {
+        const configObj = {
+            apiVersion: 'v1',
+            kind: 'Config',
+            clusters: this.clusters.map(exportCluster),
+            users: this.users.map(exportUser),
+            contexts: this.contexts.map(exportContext),
+            preferences: {},
+            'current-context': this.getCurrentContext(),
+        };
+
+        return JSON.stringify(configObj);
```

Comment:
Is there a reason we are exporting the data as json and not an object? Just wondering if people would want the json or maybe rather use yaml?

I didn't look at the go implementation though

Reply:
JSON is valid YAML :)

The purpose of this is to serialize it to a string. I can change the name to 'SerializeConfig' if that's clearer.

Reply:
Ah yes, I totally forgot. 

I was just thinking most people would use this to write their kubeconfig which is currently in `yaml` and if their intention was to convert it to `yaml` to be human-readable they would need to parse this and then convert it anyway.